hw/ip/otbn/doc/_index.md

```diff
@@ -0,0 +1,2525 @@
+      <td>RW</td>
+      <td>
+        <strong>FLAGS</strong>.
+        Wide arithmetic flags.
+        This CSR provides access to the flags used in wide integer arithmetic.
+        <table>
+          <thead>
+            <tr><th>Bit</th><th>Description</th></tr>
+          </thead>
+          <tbody>
+            <tr><td>0</td><td>Carry of Flag Group 0</td></tr>
+            <tr><td>1</td><td>LSb of Flag Group 0</td></tr>
+            <tr><td>2</td><td>MSb of Flag Group 0</td></tr>
+            <tr><td>3</td><td>Zero of Flag Group 0</td></tr>
+            <tr><td>4</td><td>Carry of Flag Group 1</td></tr>
+            <tr><td>5</td><td>LSb of Flag Group 1</td></tr>
+            <tr><td>6</td><td>MSb of Flag Group 1</td></tr>
+            <tr><td>7</td><td>Zero of Flag Group 1</td></tr>
+          </tbody>
+        </table>
+      </td>
+    </tr>
+    <tr>
+      <td>0x7D0</td>
+      <td>RW</td>
+      <td>
+        <strong>MOD0</strong>.
+        Bits [31:0] of the modulus operand, used in the BN.ADDM/BN.SUBM instructions.
+        This CSR is mapped to the RND WSR.
+      </td>
+    </tr>
+    <tr>
+      <td>0x7D1</td>
+      <td>RW</td>
+      <td>
+        <strong>MOD1</strong>.
+        Bits [63:32] of the modulus operand, used in the BN.ADDM/BN.SUBM instructions.
+        This CSR is mapped to the RND WSR.
+      </td>
+    </tr>
+    <tr>
+      <td>0x7D2</td>
+      <td>RW</td>
+      <td>
+        <strong>MOD2</strong>.
+        Bits [95:64] of the modulus operand, used in the BN.ADDM/BN.SUBM instructions.
+        This CSR is mapped to the RND WSR.
+      </td>
+    </tr>
+    <tr>
+      <td>0x7D3</td>
+      <td>RW</td>
+      <td>
+        <strong>MOD3</strong>.
+        Bits [127:96] of the modulus operand, used in the BN.ADDM/BN.SUBM instructions.
+        This CSR is mapped to the RND WSR.
+      </td>
+    </tr>
+    <tr>
+      <td>0x7D4</td>
+      <td>RW</td>
+      <td>
+        <strong>MOD4</strong>.
+        Bits [159:128] of the modulus operand, used in the BN.ADDM/BN.SUBM instructions.
+        This CSR is mapped to the RND WSR.
+      </td>
+    </tr>
+    <tr>
+      <td>0x7D5</td>
+      <td>RW</td>
+      <td>
+        <strong>MOD5</strong>.
+        Bits [191:160] of the modulus operand, used in the BN.ADDM/BN.SUBM instructions.
+        This CSR is mapped to the RND WSR.
+      </td>
+    </tr>
+    <tr>
+      <td>0x7D6</td>
+      <td>RW</td>
+      <td>
+        <strong>MOD6</strong>.
+        Bits [223:192] of the modulus operand, used in the BN.ADDM/BN.SUBM instructions.
+        This CSR is mapped to the RND WSR.
+      </td>
+    </tr>
+    <tr>
+      <td>0x7D7</td>
+      <td>RW</td>
+      <td>
+        <strong>MOD7</strong>.
+        Bits [255:224] of the modulus operand, used in the BN.ADDM/BN.SUBM instructions.
+        This CSR is mapped to the RND WSR.
+      </td>
+    </tr>
+    <tr>
+      <td>0xFC0</td>
+      <td>R</td>
+      <td>
+        <strong>RND</strong>.
+        A random number.
```

Comment:
Should we say explicitly this is different from (or related to) the RND WSR?

Reply:
Yes, we should say something. I'm just not quite sure what to say.

My thinking was the RND WSR and CSR are the same, i.e. a different view into the same LFSR/random number data output. This is effectively what we get in dcrypto, where there is only one RND data source used for control and data.

Obviously, only one register can be used at any given time, either through a 32b instruction, xor through a BN instruction.

Depending on the quality of the randomness, I can see reasons to use different random sources. We have not yet discussed the properties of RND any further, but there's an enhancement proposal tracking that at https://docs.google.com/document/d/1rgtPlXiJo21dq7gL_QZdV2XPISR78hkoLhJV8f1pt5M/edit#heading=h.k9tb5f5zh7q5.

For the purpose of pushing this document across the finish line, what's your thinking on that?

- Yes, the two RNDs can be the same and we don't need to discuss that much further. Let's put that statement into the v0.1 spec.
- Not sure, could be either way. Leave it unspecified or put a TODO in.
- They absolutely need to be different. Let's put that into the spec.

Reply:
I think it's either the first or second option, so probably the second one for now, and if we find it's the first, we can clarify later.